### ISSUE TITLE
Updating map field results in ArrayIndexOutOfBoundsException

### ISSUE THREAD

**Issue by larslorenzen** [mongodb/morphia#1118]
When i try a 'set' udate on an array field then i get the following exception:
```
Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: 1
	at java.util.Arrays$ArrayList.get(Arrays.java:3841)
	at org.mongodb.morphia.internal.PathTarget.next(PathTarget.java:96)
	at org.mongodb.morphia.internal.PathTarget.resolve(PathTarget.java:115)
	at org.mongodb.morphia.internal.PathTarget.getTarget(PathTarget.java:90)
	at org.mongodb.morphia.query.UpdateOpsImpl.add(UpdateOpsImpl.java:279)
	at org.mongodb.morphia.query.UpdateOpsImpl.set(UpdateOpsImpl.java:223)
	at de.dreamit.persistence.UpdateMapTest.main(UpdateMapTest.java:34)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:147)
```

This is the test code:
```
public class UpdateMapTest {

    public static void main(String[] args) {
        MongoClient client = new MongoClient("localhost");
        Morphia morphia = new Morphia();
        morphia.map(Settings.class);

        Datastore datastore = morphia.createDatastore(client, "test");

        Settings settings = new Settings();
        settings.properties = new HashMap<>();
        datastore.save(settings);

        Map<String, String> properties = new HashMap<>();
        properties.put("key", "value");
        UpdateOperations<Settings> updateOperations = datastore.createUpdateOperations(Settings.class);
        updateOperations.set("properties", properties);
        datastore.update(settings, updateOperations);

    }

    @Entity
    public static class Settings {
        @Id
        private ObjectId id;

        private Map<String, String> properties;

    }
}
```

**Comment by evanchooly** [comment]
This looks like a duplicate of a bug fixed in 1.3.1 which I was just about to push.  Can you try this with 1.3.1-SNAPSHOT and see if it fixes your problem?

**Comment by evanchooly** [comment]
For me, it just ran with no errors on the 1.3.x branch so it seems you have indeed hit the fixed bug.

**Comment by larslorenzen** [comment]
Great to hear that it is already fixed, i will give it a try once you pushed your changes

**Comment by evanchooly** [comment]
Duplicate of #1078 fixed in 1.3.1 which was just released.
